test/controllers/returns/credits/ExportedCreditsWeightControllerSpec.scala

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.returns.credits
+
+import base.utils.JourneyActionAnswer
+import connectors.CacheConnector
+import controllers.BetterMockActionSyntax
+import controllers.actions.JourneyAction
+import forms.returns.credits.ExportedCreditsWeightFormProvider
+import models.Mode.NormalMode
+import models.UserAnswers
+import models.UserAnswers.SaveUserAnswerFunc
+import models.requests.DataRequest
+import navigation.ReturnsJourneyNavigator
+import org.mockito.Answers
+import org.mockito.ArgumentMatchers.{eq => meq}
+import org.mockito.ArgumentMatchersSugar.any
+import org.mockito.MockitoSugar.{reset, verify, when}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatestplus.mockito.MockitoSugar.mock
+import org.scalatestplus.play.PlaySpec
+import pages.returns.credits.ExportedCreditsWeightPage
+import play.api.data.Form
+import play.api.data.Forms.{ignored, longNumber}
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, Call}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{GET, await, contentAsString, defaultAwaitTimeout, status, stubMessagesControllerComponents}
+import play.twirl.api.{Html, HtmlFormat}
+import queries.{Gettable, Settable}
+import views.html.returns.credits.ExportedCreditsWeightView
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class ExportedCreditsWeightControllerSpec extends PlaySpec with JourneyActionAnswer with BeforeAndAfterEach{
+
+  private val form = mock[Form[Long]]
+  private val dataRequest    = mock[DataRequest[AnyContent]](Answers.RETURNS_DEEP_STUBS)
+  val saveFunction = mock[SaveUserAnswerFunc]
+  val ans = mock[UserAnswers]
```

Comment:
"userUnswers"?

Reply:
yes, because in the submit function when set, save, etc they all return new object